src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/ToplevelBuilder.java

```diff
@@ -208,7 +213,17 @@ private void emitLibraries(String[] libraryNames) {
             }
             decrAlign();
             indent();
-            append("});\n\n");
+            append("});\n");
+            decrAlign();
+            indent();
+            append("}\n\n");
+            decrAlign();
+        }
+
+        private void emitInit() {
```

Comment:
Following offline discussion, I think we can simplify this by removing library loading logic from header classes and sticking it in runtime helper.